### PR TITLE
LPS-169522 Fix sorting controls and add permission customization

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

https://issues.liferay.com/issues/browse/LPS-169522

The sorting selector in the management toolbar was not correctly initialized, so it magically disappeared when entering the images tab.

Also, the blogs images tab had some custom permission checking: if a user had permissions to add blog entries, then deletion was allowed independently of the file entry permissions. To allow for this, I've added a new attribute that receives an (optional) model resource permission to use for file entries.

# How do I test it?

This issues were discovered while testing, so follow the steps described in https://issues.liferay.com/browse/LPS-169384.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
This depends on https://github.com/liferay-lima/liferay-portal/pull/3747. Putting on hold until that pr passes.

**Comment by adolfopa** [comment]
Rebased and push-forced.

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
Rebased, semver and fixed test failure.

File: modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/internal/display/context/RepositoryBrowserTagDisplayContext.java [hunk 1]
```diff
@@ -527,7 +527,8 @@ private SearchContainer<Object> _getDLSearchContainer()
 				_httpServletRequest, searchContainer.getOrderByColParam()));
 		searchContainer.setOrderByType(
 			ParamUtil.getString(
```
Comment (AliciaGarciaGarcia) [inline modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/internal/display/context/RepositoryBrowserTagDisplayContext.java:529]: On the old jsp it put to default this too:

`String orderByCol = ParamUtil.getString(request, "orderByCol", "title");`

**Review by AliciaGarciaGarcia — COMMENTED** [review]
Hi @adolfopa 

I found some issues that can or can not be related with this ticket

When you make a search on the management toolbar, it happens 2 things:

1) the order by is always reseting to modified Date
2) the sorting is not happening (maybe by the first issue)

https://user-images.githubusercontent.com/47524751/204252958-7965b256-7541-4365-830c-86cb41a73d13.mp4

It can be related or not or fixed on another issue.


on another hand, on the old way (without the FF) 

There was others order by filters: (the size filter)

old:
image


new : 
image

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia I totally agree with your suggestions. As the tests are passing, I'm going to forward this PR and send a followup fixing those issues. All this is behind a feature flag, so it can be sent safely.

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**